citesphere/src/main/java/edu/asu/diging/citesphere/web/user/DeleteCitationController.java

```diff
@@ -0,0 +1,39 @@
+package edu.asu.diging.citesphere.web.user;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.Authentication;
+import org.springframework.social.zotero.api.ItemDeletionResponse;
+import org.springframework.social.zotero.exception.ZoteroConnectionException;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import edu.asu.diging.citesphere.core.exceptions.GroupDoesNotExistException;
+import edu.asu.diging.citesphere.core.exceptions.ZoteroHttpStatusException;
+import edu.asu.diging.citesphere.core.service.ICitationManager;
+import edu.asu.diging.citesphere.user.IUser;
+
+@Controller
+public class DeleteCitationController {
+
+    @Autowired
+    private ICitationManager citationManager;
+
+    @RequestMapping(value = "/auth/group/{groupId}/references/delete", method = RequestMethod.POST)
+    public String delete(Authentication authentication, @PathVariable("groupId") String groupId, @RequestParam(value="citationList", required=false) List<String> citationList)
+            throws ZoteroConnectionException, ZoteroHttpStatusException, GroupDoesNotExistException {
+        List<ItemDeletionResponse> zoteroResponse = new ArrayList<ItemDeletionResponse>();
+        zoteroResponse = citationManager.deleteCitations((IUser) authentication.getPrincipal(), groupId, citationList);
+        for (ItemDeletionResponse response: zoteroResponse) {
+            if (response.toString() != "SUCCESS") {
+                throw new ZoteroHttpStatusException("Could not delete items. Error: " + response);
```

Comment:
this will just show an error page to the user, which is not very user-friendly. also, if the first couple got deleted but only the last one didn't, this would also not be accurate. Show an appropriate error message to the user on the items page.

Reply:
This should return a proper responseentity with a proper error code